File: C/output.c
/*
 *  Output functions for C program to calculate π using quadrature.
 *
 *  Copyright © 2008–2011, 2013  Russel Winder
 */

#include <stdio.h>

#include "output.h"

void out(char const *const banner, double const pi, int const n, double const elapseTime) {
  printf("==================== %s\n", banner);
  printf("\tpi = %.18lf\n", pi);
  printf("\titeration count = %d\n" ,  n ) ;
  printf("\telapse time = %lf\n" , elapseTime ) ;
}

void outn(char const *const banner, double const pi, int const n, double const elapseTime, int const threadCount, int const processorCount) {
  char buffer[256];
  sprintf(buffer, "%s -- thread count: %d", banner, threadCount);
  out(buffer, pi, n, elapseTime);
  printf("\tprocessor count = %d\n", processorCount);
}

File: Haskell/Output.hs
--  Output functions for the Haskell implementation of π by Quadrature.
--
--  Copyright © 2009–2011, 2013–2015  Russel Winder

module Output where

import Data.Time.Clock (getCurrentTime, diffUTCTime)
import GHC.Conc (numCapabilities)

out :: String -> Double -> Int -> IO()
out banner pi n = do
  putStrLn ("==================== " ++ show banner)
  startTime <- getCurrentTime
  putStrLn ("\tπ = " ++ show pi)
  putStrLn ("\titeration count = " ++ show n)
  endTime <- getCurrentTime
  putStrLn ("\telapse time = " ++ show (diffUTCTime endTime startTime))

outn :: String -> Double -> Int -> Int -> IO()
outn banner pi n itemCount = do
  out (banner ++ ": item count: " ++ show itemCount) pi n
  putStrLn ("\tprocessor count = " ++ show numCapabilities)
